adapters/grid/grid.go

```diff
@@ -282,6 +300,35 @@ func setImpExtData(imp openrtb2.Imp) openrtb2.Imp {
 	return imp
 }
 
+func fixNative(req json.RawMessage) (json.RawMessage, error) {
+	var gridReq map[string]interface{}
+	var parsedRequest map[string]interface{}
+
+	if err := json.Unmarshal(req, &gridReq); err != nil {
+		return req, nil
+	}
+	if imps, exists := maputil.ReadEmbeddedSlice(gridReq, "imp"); exists {
+		for _, imp := range imps {
+			if gridImp, ok := imp.(map[string]interface{}); ok {
+				native, hasNative := maputil.ReadEmbeddedMap(gridImp, "native")
+				if hasNative {
+					request, hasRequest := maputil.ReadEmbeddedString(native, "request")
+					if hasRequest {
+						delete(native, "request")
+						if err := json.Unmarshal([]byte(request), &parsedRequest); err == nil {
+							native["request_native"] = parsedRequest
+						} else {
+							native["request_native"] = request
```

Comment:
At this point you want to set native.request back even if `err!=nil`. What data other than do json you expect here? 
Here you want to replace `imp.native.request` to `imp.native.request_native`, that is not a part of openRTB, right?

Reply:
That's right, but even if we got errors when trying to parse json , we still want to get native.request to be able to see what the value is

Reply:
Thank you for the reply. 
Line 302 is not covered by unit tests and the way to do this - is to add test similar to "simple-native.json" and set `req.imp[].native.request` to string like "hello".  It will throw the Unmarshal error and L304 will be executed. 
Please add this kind of test.